internal/gatewayapi/route.go

```diff
@@ -1378,21 +1383,22 @@ func (t *Translator) processDestination(name string, backendRefContext BackendRe
 	if backendRef.Weight != nil {
 		weight = uint32(*backendRef.Weight)
 	}
-
-	backendNamespace := NamespaceDerefOr(backendRef.Namespace, route.GetNamespace())
-	err = t.validateBackendRef(backendRefContext, route, resources, backendNamespace, routeType)
-	{
-		// return with empty endpoint means the backend is invalid and an error to fail the associated route.
-		if err != nil {
-			return nil, err
-		}
-	}
-
 	// Skip processing backends with 0 weight
 	if weight == 0 {
 		return nil, nil
 	}
 
+	backendNamespace := NamespaceDerefOr(backendRef.Namespace, route.GetNamespace())
```

Comment:
why was this moved, this makes an invalid backend with weight 0 be considered as valid and will probably not publish status for it

Reply:
fixed